test/Microsoft.ML.Tests/TermEstimatorTests.cs

```diff
@@ -0,0 +1,205 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Runtime.Api;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.Model;
+using Microsoft.ML.Runtime.RunTests;
+using Microsoft.ML.Runtime.Tools;
+using System;
+using System.IO;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.ML.Tests
+{
+    public class TermEstimatorTests : TestDataPipeBase
+    {
+        public TermEstimatorTests(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        class TestClass
+        {
+            public int A;
+            public int B;
+            public int C;
+        }
+
+        class TestClassXY
+        {
+            public int X;
+            public int Y;
+        }
+
+        class TestClassDifferentTypes
+        {
+            public string A;
+            public string B;
+            public string C;
+        }
+
+
+        class TestMetaClass
+        {
+            public int NotUsed;
+            public string Term;
+        }
+
+        [Fact]
+        void TestWorking()
+        {
+            var data = new[] { new TestClass() { A = 1, B = 2, C = 3, }, new TestClass() { A = 4, B = 5, C = 6 } };
+            var xydata = new[] { new TestClassXY() { X = 10, Y = 100 }, new TestClassXY() { X = -1, Y = -100 } };
+            var stringData = new[] { new TestClassDifferentTypes { A = "1", B = "c", C = "b" } };
+            using (var env = new TlcEnvironment())
+            {
+                var dataView = ComponentCreation.CreateDataView(env, data);
+                var pipe = new TermEstimator(env, new[]{
+                    new TermTransform.ColumnInfo("A", "TermA"),
+                    new TermTransform.ColumnInfo("B", "TermB"),
+                    new TermTransform.ColumnInfo("C", "TermC")
+                });
+                var invalidData = ComponentCreation.CreateDataView(env, xydata);
+                TestEstimatorCore(pipe, dataView, null, invalidData);
```

Comment:
>TestEstimatorCore [](start = 16, length = 17)

do another one with stringData. Or make TestEstimatorCore accept an array of `invalidData`'s